src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -28,488 +29,563 @@
+						CrawlerTool.addUrlHd(f, urlHD, "");
+					}
+					if (!subtitle.isEmpty()) {
+						CrawlerTool.addUrlSubtitle(f, subtitle);
+					}
+					addFilm(f);
+				} else {
+					filmSuchen_old(urlSendung, thema, titel, dauer, datum, zeit);
+					// MSLog.fehlerMeldung(784512369, "keine URL: " + urlFilm);
+				}
+			} catch (Exception ex) {
+				Log.errorLog(762139874, ex);
+			}
+		}
+
+		private String getUrl(MSStringBuilder seite, int q) {
+			String ret = "";
+
+			seite.extractList("\"_quality\":2,\"_stream\":[", "]", liste);
+			for (String s : liste) {
+				// "http://mvideos.daserste.de/videoportal/Film/c_550000/557945/format653790.mp4","http://mvideos.daserste.de/videoportal/Film/c_550000/557945/format653793.mp4"
+				s = s.replace("\"", "");
+				if (s.contains(",")) {
+					String[] ar = s.split(",");
+					for (String ss : ar) {
+						if (ss.startsWith("http")) {
+							ret = ss;
+						}
+					}
+				}
+				if (!ret.isEmpty()) {
+					break;
+				}
+			}
+			// if (!ret.isEmpty()) {
+			// System.out.println("gefunden!!");
+			// }
+			liste.clear();
+			return ret;
+		}
+
+		private void filmSuchen_old(String urlSendung, String thema, String titel, long dauer, String datum,
+				String zeit) {
+			// für ganz alte Sachen:
+			/*
+			 * </li><li
+			 * data-ctrl-21282662-28534346-trigger="{&#039;xt_obj&#039;:{&#039;
+			 * href&#039;
+			 * :&#039;http://cdn-vod-ios.br.de/i/mir-live/bw1XsLzS/bLQH/
+			 * bLOliLioMXZhiKT1/uLoXb69zbX06/MUJIuUOVBwQIb71S/
+			 * bLWCMUJIuUOVBwQIb71S/_2rp9U1S/_-J
+			 * S/_-4y5H1S/bc3757a9-1cfd-4aaa-a963-51e9fd0095f2_,0,A,B,E,C,.mp4.
+			 * csmil/master.m3u8?__b__=200&#039;,&#039;target&#039;:&#039;_self&
+			 * #039;},&#039;redirect&#0 39;:{&#039;name&#039;:&#039;br&#039;}}">
+			 */
+			try {
+				meldung(urlSendung);
+				seite2 = getUrl.getUri_Utf(SENDERNAME, urlSendung, seite2, "");
+				if (seite2.length() == 0) {
+					Log.errorLog(612031478, "Leere Seite: " + urlSendung);
+					return;
+				}
+				String url = seite2.extract("</li><li data-ctrl-", "http://", ".m3u8");
+				if (!url.isEmpty()) {
+					url = "http://" + url + ".m3u8";
+					// System.out.println(url);
+				}
+				if (!url.isEmpty()) {
+					String beschreibung = beschreibung(urlSendung);
+					DatenFilm f = new DatenFilm(SENDERNAME, thema, urlSendung, titel, url, ""/* urlRtmp */, datum, zeit,
+							dauer, beschreibung);
+					addFilm(f);
+				} else {
+					Log.errorLog(974125698, "keine URL: " + urlSendung);
+				}
+			} catch (Exception ex) {
+				Log.errorLog(102054784, ex);
+			}
+		}
+
+		private String beschreibung(String strUrlFeed) {
+			seite3 = getUrl.getUri_Utf(SENDERNAME, strUrlFeed, seite3, "");
+			if (seite3.length() == 0) {
+				Log.errorLog(784512036, "Leere Seite: " + strUrlFeed);
+				return "";
+			}
+			return seite3.extract("<p class=\"subtitle\">", "<p class=\"teasertext\" itemprop=\"description\">", "<");
+		}
+
+	}
+
+	private boolean isAllTextsEmpty(String... aTexts) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this method into "ThemaLaden". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3398)